agent/agent.go

```diff
@@ -559,21 +594,38 @@ func updateStructValuesHelper(pluginPtr reflect.Value, newConfig map[string]inte
 	return plugin, map[string]interface{}{}, fmt.Errorf("could not update plugin")
 }
 
+func (a *Agent) CreateInput(name string) (interface{}, error) {
```

Comment:
Why are we returning an ```interface{}``` for both of these? Also what are these creating? It just seems like it's running a plugin?